src/server/scripts/Outland/GruulsLair/boss_gruul.cpp

```diff
@@ -103,7 +103,8 @@ class boss_gruul : public CreatureScript
                     case EVENT_CAVE_IN:
                         if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
                             me->CastSpell(target, SPELL_CAVE_IN, false);
-                        _caveInTimer -= 1500;
+                        if (_caveInTimer >= 4000)
```

Comment:
When you said you added a minimum of 2.5 Seconds cooldown on cave in, did you mean 4 seconds?

Reply:
It only takes off time if the current cooldown is 4 seconds or more, so the final one is a bit less than 4. Working through it, with my code the minimum is 3.5 seconds.

Cast Number | Cooldown before Cast (s) | Total time in Fight (s)
------------ | ------------- | -------------
1 | 29 | 29
2 | 27.5 | 56.5
3 | 26 | 82.5
4 | 24.5 | 107
5 | 23 | 130
6 | 21.5 | 151.5
7 | 20 | 171.5
8 | 18.5 | 190
9 | 17 | 207
10 | 15.5 | 222.5
11 | 14 | 236.5
12 | 12.5 | 249
13 | 11 | 260
14 | 9.5 | 269.5
15 | 8 | 277.5
16 | 6.5 | 284
17 | 5 | 289
18 | 3.5 | 292.5



I had a very quick look for videos or anything on the exact timing but since he casts it without an emote and the cave in can be anywhere in his arena it's hard to find anything concrete.

Reply:
👍